packages/components/src/components/editor/condition-builder/conditionBuilder.tsx

```diff
@@ -0,0 +1,80 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Button } from '@/components/common/button/button';
+import { useEffect, useState } from 'react';
+import type { Browser } from '@/components/editor/browser/browser';
+import {
+  ConditionBuilderProvider,
+  useConditionBuilderContext,
+  type ConditionBuilderProviderProps
+} from '@/components/editor/condition-builder/conditionBuilderContext';
+import { ConditionGroup } from './conditionGroup';
+
+interface ConditionBuilderProps {
+  onChange: (value: string) => void;
+}
+
+type ConditionBuilderHookProps = Omit<ConditionBuilderProviderProps, 'children'>;
+
+const conditionModes = [
+  { label: 'Basic Condition', value: 'basic-condition' },
+  { label: 'Nested Condition', value: 'nested-condition' },
+  { label: 'Always True', value: 'always-true' },
+  { label: 'Always False', value: 'always-false' }
+];
+export type ConditionMode = (typeof conditionModes)[number]['value'];
+
+export const useConditionBuilder = (props: ConditionBuilderHookProps): Browser => {
+  const [value, setValue] = useState<string>();
+
+  return {
+    name: 'Condition',
+    browser: (
+      <ConditionBuilderProvider {...props}>
+        <ConditionBuilder onChange={setValue} />
+      </ConditionBuilderProvider>
+    ),
+    applyModifier: () => {
+      return { value: value || '' };
+    },
+    infoProvider: () => <p>{value}</p>,
+    icon: IvyIcons.Process
+  };
+};
+useConditionBuilder.displayName = 'useConditionBuilder';
```

Comment:
why do you add a displayName to a hook? I think this makes no sense or am I wrong?
The displayName is a react thing to name components. E.g. they will be shown like that in the react dev tools.
So it is nice if the components have such a displayName, but I think this is useless on hooks.